frappe/public/js/frappe/ui/filters/filter.js

```diff
@@ -148,6 +148,14 @@ frappe.ui.Filter = class {
 	make_field(df, old_fieldtype) {
 		let old_text = this.field ? this.field.get_value() : null;
 
+		if(df.fieldtype=="DateRange") {
```

Comment:
Let's make this generic for all fieldtypes. Maintain a map of invalid conditions for each fieldtype. Then hide the conditions based on the map.

For e.g:
```
let invalid_condition_map = {
  Date: ['like', 'not like'],
  DateTime: ['like', 'not like'],
  Data: ['Between'],
  ...
}
```